src/main/groovy/edu/oregonstate/mist/cardsapi/resources/CardsResource.groovy

```diff
@@ -160,12 +202,48 @@ class CardsResource extends Resource {
         }
         Integer id = cardDAO.getNextId()
         cardDAO.postCard(id, (Card)newResultObject.data.attributes)
+        created(cardResultById(id)).build()
+    }
+
+    /**
+     * Endpoint for updating a Card using PUT
+     *
+     * @param id Path ID of Card to be updated
+     * @param newResultObject Contents of PUT body containing updated fields of Card
+     * @return Response
+     */
+    @PUT
+    @Path('{id}')
+    Response putCard(@PathParam('id') IntParam id, @Valid ResultObject newResultObject) {
+
+        if(!cardDAO.cardExists(id.get())) {
```

Comment:
Right, the getCardById method could be used here instead. My reasoning for using the cardExists method was that it would avoid having to do all of the computation for building a Card object since the PUT method just needs to know if the card exists rather than gather all of the data for it. I could definitely substitute for the getCardById method and delete the cardExists methods though.

Reply:
No, you don't need to change anything. I'm totally fine with your current method. Actually, this is a strong enough and pretty good reason to me. Nice 👍